src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1096,6 +1097,15 @@ private Conversion ClassifyImplicitBuiltInConversionFromExpression(BoundExpressi
                     return Conversion.ObjectCreation;
             }
 
+            // Neither Span<T>, nor ReadOnlySpan<T> can be wrapped into a Nullable<T>, therefore, there is no point to check for an attempt to convert to Nullable types here. 
+            if ((destination.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.AllIgnoreOptions) ||
+                 destination.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions)) &&
```

Comment:
This condition seems relatively expensive, at least compared to the next condition. Consider checking `source?.HasInlineArrayAttribute(out _) == true` first.

Reply:
> This condition seems relatively expensive, at least compared to the next condition. Consider checking ```source?.HasInlineArrayAttribute(out _) == true``` first.

The order is intentional. Checking ```source?.HasInlineArrayAttribute(out _) == true``` first causes a cycle. I do not plan to address it in context of this PR. Besides, symbols for well known types are cached and, since we are comparing definitions, this condition shouldn't have noticeable impact on performance.